### PR TITLE
0325 Contact CTA

### PR DESCRIPTION
**Pull request by chris-lawton**
[Link to Ticket](https://torchbox.monday.com/boards/1192293412/pulses/1347770325)

### Description of Changes Made

Styles the Contact CTA streamfield that contains avatar, text, person and CTA.

Also some small bits of housekeeping on the components template in the pattern library.

### How to Test

In the Pattern Library:
- http://localhost:8000/pattern-library/render-pattern/patterns/molecules/streamfield/blocks/contact_call_to_action.html
- http://localhost:8000/pattern-library/render-pattern/patterns/styleguide/components/components.html
- http://localhost:8000/pattern-library/render-pattern/patterns/pages/standard/standard_page.html

Or by creating a Contact CTA streamfield on a standard page on the site.

### Screenshots

<details>
  <summary>Expand to see more</summary>

Screenshot 2024-01-22 at 16 34 28

Screenshot 2024-01-22 at 16 34 34

</details>

### MR Checklist

- [x] Add a description of your pull request and instructions for the reviewer to verify your work.
- [x] If your pull request is for a specific ticket, link to it in the description.
- [x] Stay on point and keep it small so the merge request can be easily reviewed.
- [ ] Tests and linting passes.

#### Unit tests

- [ ] Added
- [ ] Not required

#### Documentation

- [ ] Updated build docs
- [ ] Updated editor guidelines (https://docs.google.com/document/d/1PAWccdQ4tfaZsrEWmpDhvP3GH5RRmBOARFVp4b-kje8/edit?usp=sharing)
- [x] Not required

#### Browser testing

- [x] I have tested in the following browsers and environments (edit the list as required)
  - Latest version of Chrome on mac
  - Latest version of Firefox on mac
  - Latest version of Safari on mac
  - Safari on last two versions of iOS
  - Chrome on last two versions of Android
- [ ] Not required

#### Data protection

- [ ] Not relevant
- [ ] This adds new sources of PII and documents it and modifies Birdbath processors accordingly

#### Accessibility

- [x] Automated WCAG 2.1 tests pass
- [x] Manual WCAG 2.1 tests completed
- [x] I have tested in a screen reader
- [x] I have tested in high-contrast mode
- [ ] Any animations removed for prefers-reduced-motion
- [ ] Not required

#### Sustainability

- [ ] Images are optimised and lazy-loading used where appropriate
- [ ] SVGs have been optimised
- [ ] Perfomance and transfer of data considered
- [ ] If JavaScript is needed alternatives have been considered
- [ ] New font variants have not been added
- [x] Not required

#### Pattern library

- [x] The pattern library component for this template displays correctly, and does not break parent templates
- [ ] The styleguide is updated if relevant
- [ ] Changes are not relevant the pattern library

### CODE REVIEW

File: tbx/project_styleguide/templates/patterns/molecules/streamfield/blocks/contact_call_to_action.html [hunk 1]
```diff
@@ -19,6 +18,11 @@
                 <span class="contact-cta__role">{{ value.person.role }}</span>
             {% endif %}
         </cite>
-
     {% endif %}
+
+    {% for cta in value.call_to_action %}
```
Comment (helenb) [inline tbx/project_styleguide/templates/patterns/molecules/streamfield/blocks/contact_call_to_action.html:23]: Do you know why there is a forloop here? You can only add one button to a cta.

File: tbx/project_styleguide/templates/patterns/molecules/streamfield/blocks/contact_call_to_action.yaml [hunk 2]
```diff
@@ -1,21 +1,23 @@
...
+      name: Olly Willans
+      role: Chief People and Planet Officer
 
 tags:
   image:
-    avatar width-143 format-webp class="avatar__image":
-      # target_var: person_image
-      raw: |
-        <img src="https://source.unsplash.com/143x143?face" width="143" height="143" alt="Some alt" class="avatar__image">
+    avatar fill-144x144 format-webp as avatar_desktop:
```
Comment (helenb) [inline tbx/project_styleguide/templates/patterns/molecules/streamfield/blocks/contact_call_to_action.yaml:14]: Do you think it is worth adding a different image variant for the small avatar at desktop? I think perhaps the extra code needed isn't worth it but interested if you had any thoughts.

File: tbx/static_src/sass/components/_contact-cta.scss [hunk 3]
```diff
@@ -1,38 +1,27 @@
...
-        margin-bottom: $spacer-small;
-    }
+.contact-cta {
+    display: grid;
+    grid-template-columns: repeat(12, 1fr);
 
     &__text {
-        font-style: normal;
-        margin-bottom: $spacer-small;
+        grid-column: 1 / span 11;
```
Comment (helenb) [inline tbx/static_src/sass/components/_contact-cta.scss:8]: I like that the child elements of this component are aligned with the grid. I didn't really manage that with the listing items when I probably should have done, but the way the elements needed to be nested meant it didn't quite work.

**Review by helenb — APPROVED** [review]
Thanks @chris-lawton this looks good. Main query is about why we have a loop around the buttons, which I appreciate maybe came from the back-end code.

File: tbx/static_src/sass/components/_contact-cta.scss (hunk 3 above)
Comment (chris-lawton) [reply tbx/static_src/sass/components/_contact-cta.scss:8]: Yeah it's a tricky one as it's a nested grid. I had to go down that route due to the layout differences between desktop and mobile and without duplicating elements and showing/hiding for mobile/desktop.

File: tbx/project_styleguide/templates/patterns/molecules/streamfield/blocks/contact_call_to_action.yaml (hunk 2 above)
Comment (chris-lawton) [reply tbx/project_styleguide/templates/patterns/molecules/streamfield/blocks/contact_call_to_action.yaml:14]: Hmm, that's a good question. I think i'd agree that the extra code probably isn't worth it as the image we're serving it so small. Maybe a nice to have if we get time for it : )

File: tbx/project_styleguide/templates/patterns/molecules/streamfield/blocks/contact_call_to_action.html (hunk 1 above)
Comment (chris-lawton) [reply tbx/project_styleguide/templates/patterns/molecules/streamfield/blocks/contact_call_to_action.html:23]: Hmm i'm not sure to be honest. @lparsons396 or @engineervix - can you shed any light?

Comment (lparsons396) [reply tbx/project_styleguide/templates/patterns/molecules/streamfield/blocks/contact_call_to_action.html:23]: Yeah, quite right. That's my mistake from my initial set-up of the file

Comment (chris-lawton) [reply tbx/project_styleguide/templates/patterns/molecules/streamfield/blocks/contact_call_to_action.html:23]: No probs - let's ticket it and resolve in another MR so not to hold testing up

**Pull request merged**